main/adapters/handlers/verifier.go

```diff
@@ -221,7 +223,23 @@ func (v *Verifier) loadExternalIdentityPublicKey(verifyFromKnownIdentitiesOnly b
 		return nil, err
 	}
 
-	return v.VerifierProtocol.LoadPublicKey(id)
+	pubKeyPEM, err = v.VerifierProtocol.LoadPublicKey(id)
+	if err != nil {
+		return nil, err
+	}
+
+	go func() {
```

Comment:
this method is only called in case the public key did not exist in the database. It would also result in an error to try and overwrite an existing row in the database.

Reply:
OK, then it is fine I would say.